### ISSUE TITLE
Symfony displays a blank page instead of an error page under certain circumstances

### ISSUE THREAD

**Issue by C3rise** [symfony/symfony#8703]
### How to reproduce
1. Create a controller action that contains `$foo->bar();`. Make sure that `$foo` is undefined.
2. Call said controller in both the `prod` and `dev` environments.
### Code sample

``` php
<?php

// src/Acme/BugReportBundle/Controller/BuggyController.php
namespace Acme\BugReportBundle\Controller;

class BuggyController
{
    public function indexAction()
    {
        $foo->bar();
    }
}
```
### Attempt at analysis

When `indexAction()` is called, Symfony will display a blank page if `Debug::enable()` was called before (for example, if the `dev` environment is being used).

If you insert `$foo = null;` before `$foo->bar();`, then Symfony's fatal error page will be displayed as intended.

`Debug::enable()` registers `\Symfony\Component\Debug\ErrorHandler` as an error handler. `$foo->bar()` will make PHP display two errors:
1. Notice: Undefined variable: foo in _file_ on line _line_
2. Fatal error: Call to a member function bar() on a non-object in _file_ on line _line_

The notice will trigger `ErrorHandler::handle()`, which will in turn raise a `\Symfony\Component\Debug\Exception\ContextErrorException`. However, presumably because of the fatal error in the same line, Symfony's "Exception detected" page will not be displayed. The thrown `ContextErrorException` also prevents `ErrorHandler::handleFatal()` from being called (if the line that throws said exception is commented out, `handleFatal()` will be called and the error page is displayed properly). Because of this, a blank page will be displayed which is not really useful for debugging.
### Other information

**Operating system:** Debian GNU/Linux 6.0.7 (Linux 2.6.32 i686)
**Symfony version:** 2.3.3
**PHP version:** 5.3.6-11 (running via FastCGI)

**Comment by bassemBMI** [comment]
I fixed the problem by changing the database user to the root, I think it is related to the privilege of DB user.

**Comment by yabiBoy** [comment]
try this

cache:clear --env=prod --no-debug

**Comment by leonardola** [comment]
Guys i had this same problem here with php 5.5.30 and apache 2.4.16, the solution was to just rise the amount of memory php is allowed to use. so if you could not fix it yet try rising the amount of memory on php.ini

**Comment by fawzyj** [comment]
I have the same problem, php 5.5.9 on ubuntu 14.04
upgraded php to 5.6.19 in hope it will resolve the issue but the problem persist.
I tried with symfony 2.8 and 3.0.1 with no change

Any work aroud? please let me know even if it is a temporary solution as i cannot proceed in development.

Thanks in advance.

**Comment by fawzyj** [comment]
Sorry, I cannot understand....
You are sending a cookie header, is this a solution or what? why you send this header?
What configuration should i check in php.ini ? what are the recommended values?

**Comment by fawzyj** [comment]
I updated my app dependencies to the latest symphony v3.0.4, problem persists.

**Comment by mvrhov** [comment]
@fawzyj: There is a hint above. Did you up the php memory size maybe 128M in this particular error case is not enough.

**Comment by fawzyj** [comment]
it is set to -1 which means no upper limit, the script can use any amount of memory, my machine is 8GB, i think it is enough.

**Comment by andho** [comment]
Happened to me today. Fixed after running `./bin/console cache:clear`.

Also when running `php -S 0.0.0.0:8000 ../vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/router_dev.php` from withing the web folder, symfony was able to display the error.

Another thing to note is, nothing goes into the symfony log. I didn't turn on php error log though.

**Comment by andho** [comment]
False alarm (bug not fixed). `./bin/console cache:clear` did fix the issue with `$foo->bar()`. But the real error caused by `$form->handleRequest($request);` which ended up with the following fatal error `PHP Fatal error:  Call to a member function add() on null` was not shown by Symfony.

Also note that display errors is on.

Symfony 3.1.2

php -v:
PHP 5.6.11-1ubuntu3.4 (cli) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2015, by Zend Technologies
    with Xdebug v2.3.3, Copyright (c) 2002-2015, by Derick Rethans

uname -a
4.2.0-42-generic #49-Ubuntu SMP Tue Jun 28 21:26:26 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**Comment by COil** [comment]
Hi, very weird, I am facing a similar issue, when calling a non existent method, I've got a blank page with Symfony 3.1.3 while with Symfony 2.8.9 I've got the good correct error page (in debug mode).
When triggering a notice, I've got the good debug page with both versions.  When enabling xdebug, I've got the error stack trace in the apache logs. 

And in error.log , apache crashes:
 
    [Fri Aug 19 15:56:41.950087 2016] [core:notice] [pid 3307] AH00051: child pid 16208 exit signal Segmentation fault (11), possible coredump in /etc/apache2

Any thought ?

**Comment by zhil** [comment]
I have the same problem with symfony 2.8.9 and 2.8.11 - in prod environment I just get blank page and wasnt able to catch it neither with 
```
twig:
    exception_controller:  AppBundle:Exception:showException
```
not with
```
public static function getSubscribedEvents()
    {
        // return the subscribed events, their methods and priorities
        return [
            KernelEvents::EXCEPTION => [['onKernelException', 200]],
        ];
    }
```

In my case I have upgraded vendors for old symfony2 application and my app/AppKernel.php have such a code 
```
\Monolog\ErrorHandler::register($logger);
```
After disabling that line my error listener start working :)

**Comment by The-backend-dev** [comment]
I just got this bug on symfony 3.2.8 where no error was displayed in development mode in the browser but it displayed a critical error in the logs correctly. Steps to reproduce: 
In the entity class place a wrong directory for the repository class in a symfony 3.2.8 project example:
* @ORM\Entity(repositoryClass="Backend\notExistingBundle\Repository\TagRepository")

The same error displayed correctly in a 2.8 project in an error screen.

**Comment by COil** [comment]
This issue has been closed but apparently the problem remains.

**Comment by guzinski** [comment]
I got the same bug.

Blank page when NPE occurred into the $form->handleRequest($request)

Symfony 3.2.9

**Comment by COil** [comment]
Yes still happens in 3.3.2 too. Is there a new issue for this problem has this one is quit old ? But the first message is still relevant because it 's `Debug::enable();` that leads to this weird behaviour. (+ other PHP error settings I suppose)

**Comment by d-ph** [comment]
This is related to the fact, that `Symfony\Component\Debug\Debug` disables error reporting, when the request is not run from cli [ini_set('display_errors', 0);](https://github.com/symfony/debug/blob/c5b6e26/Debug.php#L46). This is supposed to be fine, because the Debug component is registering its own error handler. The problem is that (probably: some) PHP's Fatal Errors immediately crash the script without running those custom error handlers, which leaves devs with the white page of death. One of those Fatal Errors is: 
```
Fatal error: Class FooBar contains 1 abstract method and must therefore be declared abstract or implement the remaining methods
```

**Comment by nicolas-grekas** [comment]
would you be able to provide a patched standard edition that fails that way?

**Comment by Padam87** [comment]
Hi,

@nicolas-grekas https://github.com/Padam87/symfony-standard/commit/8174987a3bafa883f94a3e5fc32924efc3177559

As expected CLI displays errors, and `ini_set('display_errors', 1);` after enabling debug fixes the issue.

**Comment by Padam87** [comment]
https://github.com/symfony/symfony/blob/7974cd74e0410143e1ad8ace605b00869d5f88ca/src/Symfony/Component/Debug/ErrorHandler.php#L463

These exceptions are not caught by the exception handler.

Even tried to set it again, with no effect. Calling `$this->handleException($errorAsException);` displays the exception.

**Comment by d-ph** [comment]
I had some time today evening to create a test case for this bug.

php 7.1.8
symfony 3.3.6
any os

Steps to reproduce:
1. `git clone --branch=bug-report/white-page-of-death https://github.com/d-ph/symfony-standard.git`
2. `cd symfony-standard\`
3. `composer install`
4. `php bin\console server:run`
5. Go to `http://127.0.0.1:8000/`. Confirm it returns a pretty error page with error:
```
Error: Class MyApp\Foo contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (MyApp\AbstractFoo::bar)
```
6. Stop the `server:run` (ctrl-c). 
7. `vim app\AppKernel.php`
8. Uncomment the `new \Sentry\SentryBundle\SentryBundle(),` line
9. `php bin\console server:run`
10. Go to `http://127.0.0.1:8000/`

Result:
HTTP500 with white page of death. 

Expected result:
The same error page as in step 5.

More info:
The issue is triggered after installing and enabling `sentry/sentry-symfony`, but I doubt that the problem lies in that package. I believe any code trying to register an error handler with `set_error_handler` *AFTER* `Symfony\Component\Debug\ErrorHandler` will trigger the problem. When it happens, then https://github.com/symfony/debug/blob/856a3e1/ErrorHandler.php#L563 :
```php
        $handler = set_error_handler('var_dump');
        $handler = is_array($handler) ? $handler[0] : null;
        restore_error_handler();
        
        if (!$handler instanceof self) {
            return;
        }
```
will no longer want to handle any errors, because the registered error handler is not `Symfony\Component\Debug\ErrorHandler`.

Xdebug screenshot, when I get the pretty error page:
![image](https://user-images.githubusercontent.com/4347095/29497956-3af8602c-85ea-11e7-99e2-5f7b7ae6eb93.png)

Xdebug screenshot, when I get the white screen of death:
![image](https://user-images.githubusercontent.com/4347095/29497957-420f15ae-85ea-11e7-948f-f1eddea2426c.png)

I bet there are at least a couple of ways to solve it, but none of them would be pretty. It's just php things again.

**Comment by COil** [comment]
@d-ph thanks for this very complete bug report.

**Comment by ostrolucky** [comment]
Here's easier way to reproduce:
```
composer create-project symfony/skeleton debug-test
cd dbg-test
echo "<?php class A implements \Serializable {}" > src/A.php
php -d display_errors=1 -d display_startup_errors=1 -d error_reporting=30719 -S 127.0.0.1:8000 -t public
curl http://127.0.0.1:8000
```

**Comment by nicolas-grekas** [comment]
Fixed in #25408

**Comment by ErvinSabic** [comment]
@fabpot 
This might have resurfaced. 

- I created a new bundle and added it to my composer.json
- Then I ran composer install
- I then began creating controllers for my bundle and noticed that all of my routes started returning blank pages. 
- I thought maybe it was a weird caching issue so I ran php bin/console cache:clear and it returned a php exception.
- Came to the conclusion that Symfony is not displaying errors. 

Also on my homepage route ( / ), I am getting a Javascript console error that says Uncaught ReferenceError: Sfjs is not defined. It only does this on that route. The other pages are fine. Although I am not sure if these issues are related. 

Dev machine specs / environment: 
OS: Linux Mint 18.2 "Sonya"
Symfony Version: 3.4.9
Symfony Environment: Dev
Apache Version: Apache/2.4.18
PHP Version: 7.2.4
Composer: version @package_branch_alias_version@ (1.0.0-beta2) 2016-03-27 16:00:34 (Holy crap)
Mysql Version: 5.7.20-0ubuntu0.16.04.1
